src/main/java/seedu/homerce/logic/commands/service/DeleteServiceCommand.java

```diff
@@ -53,4 +61,24 @@ public boolean equals(Object other) {
             || (other instanceof DeleteServiceCommand // instanceof handles nulls
             && targetIndex.equals(((DeleteServiceCommand) other).targetIndex)); // state check
     }
+
+
+    /**
+     * Checks if the service that is about to be deleted exists in Homerce's future appointments.
+     *
+     * Deletion will be prevented if the service exists in Homerce's appointments dated today or in the future.
+     * This prevents two different services with the same service code existing in Homerce's upcoming appointments.
+     */
+    private boolean isValidDeletion(Service serviceToDelete, List<Appointment> appointments) {
+        if (appointments == null) {
+            return true;
+        }
+        LocalDate today = LocalDate.now();
+        return appointments.stream().noneMatch((appointment) -> {
+            LocalDate appointmentDate = appointment.getAppointmentDate().getLocalDate();
+            Service appointmentService = appointment.getService();
+            return (appointmentDate.isAfter(today) || appointmentDate.isEqual(today))
```

Comment:
maybe can extract out some stuff here since there are two return statements and many checks for the conditional

Reply:
Got it. Will extract the check into a separate `checkValidity` method. Thanks!